src/EthereumTransactionData.js

```diff
@@ -0,0 +1,64 @@
+import CACHE from "./Cache.js";
+
+export default class EthereumTransactionData {
+    /**
+     * @protected
+     * @param {object} props
+     * @param {Uint8Array} props.callData
+     */
+    constructor(props) {
+        this.callData = props.callData;
+    }
+
+    /**
+     * @param {Uint8Array} bytes
+     * @returns {EthereumTransactionData}
+     */
+    static fromBytes(bytes) {
+        if (bytes.length === 0) {
+            throw new Error("empty bytes");
+        }
+
+        if (bytes[0] != 2) {
+            if (CACHE.ethereumTransactionDataLegacyFromBytes == null) {
+                throw new Error(
+                    "CACHE.ethereumTransactionDataLegacyFromBytes is not set"
+                );
+            }
+
+            return CACHE.ethereumTransactionDataLegacyFromBytes(bytes);
+        } else {
+            if (CACHE.ethereumTransactionDataEip1559FromBytes == null) {
+                throw new Error(
+                    "CACHE.ethereumTransactionDataLegacyFromBytes is not set"
```

Comment:
Didn't catch this before - looks like this error message is for `[...]LegacyFromBytes`, should be for `[...]Eip1559FromBytes`

Reply:
Good catch, fixed.